repo/fsrepo/fsrepo.go

```diff
@@ -102,6 +112,24 @@ func open(repoPath string) (repo.Repo, error) {
 	if !isInitializedUnsynced(r.path) {
 		return nil, debugerror.New("ipfs not initialized, please run 'ipfs init'")
 	}
+
+	// Check version, and error out if not matching
+	ver, err := ioutil.ReadFile(path.Join(expPath, "version"))
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, errors.New("version check failed, no version file found, please run 0-to-1 migration tool.")
+		}
+		return nil, err
+	}
+
+	vers := string(ver)[:1]
+
+	if vers != RepoVersion {
+		return nil, fmt.Errorf("Repo has incorrect version: '%s'\nProgram version is: '%s'\nPlease run the appropriate migration tool before continuing",
```

Comment:
var MsgUpgradeRepo = `Repo has incorrect version: %s
Program version is: %s
Please run the repo migration tool before continuing:

    <cmd for them to run>
`

I'll think about what a good way to do `<cmd for them to run>` is. the problem is they:
- may not have it installed (installed ipfs as a binary).
- may not have go installed (may not be able to compile).
- do not have ipfs installed (cannot `ipfs cat <hash>/<arch>/1-to-2 >1-to-2`)


May be able to do something like:
```sh
wget http://gateway.ipfs.io/ipfs/<hash>/<arch>/1-to-2
chmod +x 1-to-2
./1-to-2
```

And ship a little shell script that does it as:

```sh
ipfs-repo-migrate 1-to-2
```